### ISSUE TITLE
Add linear mixed effect module

### ISSUE THREAD

**Issue by raphaelvallat** [raphaelvallat/pingouin#103]
This is most likely a longer-term project. Linear mixed-effects (LME) regression has several advantages over repeated measures ANOVA and I think it would be really useful to add it to Pingouin.

The most exhaustive LME package is obviously the [lme4](https://cran.r-project.org/web/packages/lme4/index.html) package for R. In Python, statsmodels has a [dedicated class](https://www.statsmodels.org/stable/mixed_linear.html), and there is also the [pymer4](http://eshinjolly.com/pymer4/) which is based on the rpy2 interface.

Practically speaking, this will require a class implementation, as in statsmodels. We should probably start with support for simple models and build on that. May require adding a dependency for [patsy](https://patsy.readthedocs.io/en/latest/) if we want a formula-based syntax.

**Comment by GanshengT** [comment]
Thank you for your great work. I am already excited to see the new functionality. One of the assumptions of mixed_anova, 'homogeneity of variance-covariance matrices' is usually tested with the Box’s M test. In my opinion, implementing this test would make pingouin more complete. What are your thoughts?

**Comment by raphaelvallat** [comment]
Hi @GanshengT,

I agree that it would make a nice addition to Pingouin. Would you be able to open a new issue for that and/or submit a PR?

Thanks,
Raphael

**Comment by GanshengT** [comment]
Yes. I will open a new issue. It might take some time for me to go through the structure of Pingouin before submitting a PR. I'll see to it.
